event/src/unvalidated.rs

```diff
@@ -0,0 +1,249 @@
+//! Types of raw unvalidated Ceramic Events
+
+use std::collections::BTreeMap;
+
+use cid::Cid;
+use ipld_core::ipld::Ipld;
+use serde::{Deserialize, Serialize};
+
+use crate::bytes::Bytes;
+
+/// Ceramic Event
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Event {
+    /// Time event in a stream
+    Time(Box<TimeEvent>),
+    /// Signed event in a stream
+    Signed(JWS),
+}
+
+/// Signed event
+#[derive(Debug, Serialize, Deserialize)]
+pub struct JWS {
+    payload: Bytes,
+    signatures: Vec<Signature>,
+}
+
+impl JWS {
```

Comment:
why are we parsing out the link Cid and capability Cid on demand every time they are asked for?  Why doesn't the JWS struct just deserialize the data into Cids upfront so that then all callers can just access them directly when needed?

I guess this is becoming a pattern of questions for me.  I feel like we keep leaning towards very lazy approaches to deserialization but I'm worried this is going to make it very easy to wind up parsing the same data over and over again.  It feels like it would be better convention to always try to parse things at the edges once then pass around strongly-typed structures throughout all of our internal code.

Reply:
okay, I think this makes sense at least for now.  Still slightly worried about it, but we can optimize later if need be.